src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/UpgradeReadiness.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    /// <summary>
+    /// Used by IUpgradeReadyCheck instances to declare whether or not upgrade-assistant can upgrade a project.
+    /// </summary>
+    public enum UpgradeReadiness
+    {
+        /// <summary>
+        /// Upgrade-assistant is ready to help upgrade this project.
+        /// </summary>
+        Ready,
+
+        /// <summary>
+        /// This project contains something that prevents upgrade.
+        /// </summary>
+        NotReady,
+
+        /// <summary>
+        /// Upgrade-assistant can help with some of the tasks required to upgrade this project.
+        /// But users should be informed that upgrade-assistant cannot completely upgrade this project.
+        /// </summary>
+        Unsupported
```

Comment:
Super-duper nit-pick: Can we re-order these so that NotReady comes after Unsupported? My reasoning is that 'not ready' is even more unsupported than unsupported. If someone ever tried to do a comparison of enum values, it would make sense for them to get more unsupported as the list goes on.